File: 0009-palindrome-number/0009-palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    let str = String(x)

    if(str < 0) return false

    for(let i = 0; i < str.length/2; i++){
        if(str[i] == str[str.length -1 -i]) continue
        else{ return false}
    }
    return true
};

File: 0009-palindrome-number/0009-palindrome-number.kt
class Solution {
    fun isPalindrome(x: Int): Boolean {
        var reverse = 0

        if (x < 0) {
            return false
        }

        var temp = Math.abs(x)


        while(temp != 0){
            reverse = (reverse * 10) + (temp % 10)
            temp /= 10
        }

        return reverse == Math.abs(x)  
    }
}